src/main/java/frc/robot/Subsystem/Climb/ClimbSubsystem.java

```diff
@@ -26,37 +30,53 @@ public static ClimbSubsystem getInstance(){
 
   /** Creates a new ClimbSubsystem. */
   private ClimbSubsystem() {
-    m_rightMotor = new CANSparkMax(M_CLIMBINGRIGHT_MOTOR_ID, MotorType.kBrushless);
-    m_leftMotor = new CANSparkMax(M_CLIMBINGLEFT_MOTOR_ID, MotorType.kBrushless);
-    m_leftMotor.setInverted(false);
-    m_rightMotor.setInverted(true);
+    m_rightMotor = new CANSparkMax(M_CLIMBING_RIGHT_MOTOR_ID, MotorType.kBrushless);
+    m_leftMotor = new CANSparkMax(M_CLIMBING_LEFT_MOTOR_ID, MotorType.kBrushless);
+
+    configs();
+  }
+
+  public DoubleSupplier getSpeed()
+  {
+    return (() -> m_rightMotor.getAppliedOutput());
   }
```

Comment:
.getAppliedOutput() doesn't return the motor's speed.

also, you only return the speed of 1 side.

also, why use a double supplier here?

Reply:
i use double supplier because it doesn't hurt

Reply:
why do i even need to return the speed?

Reply:
> why do i even need to return the speed?

because the method is called getSpeed()

Reply:
> i use double supplier because it doesn't hurt

it is not the proper way to use it has no use and only make it more complex in the function itself and when using it

if i have a function getSpeed() that returns double and i want double supplier i can just do ()->getSpeed()

Reply:
but why do i need this func

Reply:
idk you added it not me